### PR TITLE
Creating a new tool for validating sql query

### PR DESCRIPTION
**Pull request by arunraja1**
- Inherited AgentExecutor and created a StateAgentExecutor and modified _take_next_step function which has state variable (previous tools output)
- Inherited BaseTool and created stateTool  class which has state variable and passed to the tools
- Modified the existing tools in UnityCatalog toolkit to inherit from StateTool instead of BaseTool
- Created a new tool named SqlQueryValidatorTool which takes in sql_query and schema and validates the query

### CODE REVIEW

File: libs/langchain/src/langchain/agents/__init__.py [hunk 1]
```diff
@@ -15,18 +15,11 @@
...
-                                        
-
     BaseMultiActionAgent  --> OpenAIMultiFunctionsAgent
-    
-    
 **Main helpers:**
-
 .. code-block::
-
     AgentType, AgentExecutor, AgentOutputParser, AgentExecutorIterator,
```
Comment (arunraja1) [inline libs/langchain/src/langchain/agents/__init__.py:21]: Update State Agent Executor

File: libs/langchain/src/langchain/tools/spark_unitycatalog/tool.py [hunk 2]
```diff
@@ -223,3 +232,151 @@ def get_table_list_from_unitycatalog(
...
+        query: str,
+        run_manager: Optional[CallbackManagerForToolRun] = None,
+    ) -> str:
+        """Execute the query, return the results or an error message."""
+        extracted_sql_query = self._extract_sql_query()
+        if extracted_sql_query:
+            executable_query = extracted_sql_query.strip()
+        else:
+            executable_query = query.strip()
+        if executable_query.startswith('"'):
```
[the remark this answers is not in the document]
Comment (arunraja1) [reply libs/langchain/src/langchain/tools/spark_unitycatalog/tool.py]: Pushed a commit

File: libs/langchain/src/langchain/agents/__init__.py (hunk 1 above)
Comment (arunraja1) [reply libs/langchain/src/langchain/agents/__init__.py:21]: Pushed a commit

**Review by abhilashsharma1992 — APPROVED** [review]

File: libs/langchain/src/langchain/tools/spark_unitycatalog/tool.py [hunk 3]
```diff
@@ -223,3 +232,148 @@ def get_table_list_from_unitycatalog(
...
+        raise NotImplementedError("ListSqlTablesTool does not support async")
+
+    def _setup_llm(self):
+        openai.api_type = "azure"
+        openai.api_key = os.getenv("OPENAI_API_KEY")
+        openai.api_base = os.getenv("OPENAI_API_BASE")
+        llm = AzureChatOpenAI(
+            deployment_name=os.getenv("AZURE_OPENAI_DEPLOYMENT_NAME"),
+            temperature=0,
+            openai_api_version="2023-03-15-preview",
```
[the remark this answers is not in the document]
Comment (arunraja1) [reply libs/langchain/src/langchain/tools/spark_unitycatalog/tool.py]: Pushed a commit

**Pull request merged**